packages/dynamite/dynamite/lib/src/builder/client.dart

```diff
@@ -498,8 +498,14 @@ final authentication = $client.authentications.firstWhereOrNull(
 
   yield* securityRequirements.map((final requirement) {
     final securityScheme = spec.components!.securitySchemes![requirement.keys.single]!;
+    final nameParts = <String?>[
+      securityScheme.type,
+      securityScheme.scheme,
+      securityScheme.$in,
+      securityScheme.name,
+    ].whereNotNull();
```

Comment:
I this fits better as a method in the security scheme object

Reply:
I'll make a separate PR then